proto/messages.go

```diff
@@ -385,46 +384,52 @@ func readMessageSet(r io.Reader, size int32) ([]*Message, error) {
 		_ = msgdec.DecodeInt8()
 
 		attributes := msgdec.DecodeInt8()
+
+		if version != -1 && version >= KafkaV1 {
```

Comment:
I would be more strict about the version. We don't know what versions 2+ will have.
So I'd do:
```
switch version {
case KafkaV1:
    // timestamp
    _ = msgdec.DecodeInt64()
}
```

Reply:
done